src/app/pages/system/general-settings/ntp-server/ntp-server-list/ntp-server-list.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('should show table headers', async () => {
+    const table: IxTableHarness = await loader.getHarness(IxTableHarness);
+    const headerRow = await table.getHeaderRow();
+
+    expect(Object.keys(headerRow).length).toBe(7);
+    expect(headerRow.address).toBe('Address');
+    expect(headerRow.burst).toBe('Burst');
+    expect(headerRow.iburst).toBe('IBurst');
+    expect(headerRow.prefer).toBe('Prefer');
+    expect(headerRow.minpoll).toBe('Min Poll');
+    expect(headerRow.maxpoll).toBe('Max Poll');
+    expect(headerRow.actions).toBe('');
+  });
+
+  it('should show table rows', async () => {
+    spectator.fixture.componentInstance.loading = false;
+    spectator.fixture.componentInstance.error = false;
+    spectator.fixture.componentInstance.createDataSource(fakeDataSource);
+    const table = await loader.getHarness<IxTableHarness>(IxTableHarness);
+    const cells = await table.getRowCells();
+
+    const expectedRows = [
+      {
+        actions: 'delete',
+        address: '2.debian.pool.ntp.org',
+        burst: 'false',
+        iburst: 'true',
+        prefer: 'false',
+        minpoll: '6',
+        maxpoll: '10',
+      },
+      {
+        actions: 'delete',
+        address: '1.debian.pool.ntp.org',
+        burst: 'false',
+        iburst: 'true',
+        prefer: 'false',
+        minpoll: '6',
+        maxpoll: '10',
+      },
+      {
+        actions: 'delete',
+        address: '0.debian.pool.ntp.org',
+        burst: 'false',
+        iburst: 'true',
+        prefer: 'false',
+        minpoll: '6',
+        maxpoll: '10',
+      },
+    ];
+
+    expect(ws.call).toHaveBeenCalledWith('system.ntpserver.query');
+    expect(cells).toEqual(expectedRows);
+  });
+
+  it('should show empty table', async () => {
+    spectator.fixture.componentInstance.loading = false;
+    spectator.fixture.componentInstance.error = false;
+    spectator.fixture.componentInstance.createDataSource([]);
+    const table = await loader.getHarness<IxTableHarness>(IxTableHarness);
+    const text = await table.getCellTextByIndex();
+
+    expect(text).toEqual([['No servers have been added yet']]);
+  });
+
+  it('should show error if can not retrieve response', async () => {
+    spectator.fixture.componentInstance.loading = false;
+    spectator.fixture.componentInstance.error = true;
+    spectator.fixture.componentInstance.createDataSource([]);
+    const table = await loader.getHarness<IxTableHarness>(IxTableHarness);
+    const text = await table.getCellTextByIndex();
+
+    expect(text).toEqual([['Can not retrieve response']]);
+  });
+
+  it('should open add ntp server form', async () => {
+    const addButton = await loader.getHarness(MatButtonHarness.with({ text: 'Add' }));
+    await addButton.click();
+
+    expect(spectator.fixture.componentInstance.doAdd).toHaveBeenCalledTimes(1);
+  });
+
+  it('should display confirm dialog of deleting ntp server', async () => {
```

Comment:
Since deletion of data is also something that happens in this component, we also need to make sure that proper ws call is made.

Reply:
Added the check but it does not work properly, so it is commented out until figure out a cause. Received `system.ntpserver.query` instead of expected `system.ntpserver.delete`. Even if `doDelete` does not use `dialog.confirm` and `getData()` and doing only `ws.call('system.ntpserver.delete')`

Reply:
It may have something to do with:
```
    jest.spyOn(spectator.fixture.componentInstance, 'doDelete').mockImplementation();
```
😛

Reply:
On this note, instead of testing that `doAdd` was called, let's mock `slideInService` and check that `open` was called with correct class.